application.js

```diff
@@ -0,0 +1,278 @@
+// 자주 사용되는 HTML 요소들을 저장해두기 위한 변수
+let newTodoInput = null
+let todoList = null
+let todoElements = null
+let todoListCount = 0
+let filters = null
+let filterAll = null
+let filterActive = null
+let filterCompleted = null
+let filterContainer = null
+let selectedFilter = null
+// 자주 사용되는 객체, 상수값을 저장해두기 위한 변수
+let todoElementsNameArray = null
+const ToDoElementStorage = window.localStorage
+const ENTER_KEYCODE = 13
+const ESC_KEYCODE = 27
+const KEYWORD = "gTZ5JMw51a"
+
+// 웹페이지 로드 시 실행되는 함수
+function init(){
+    // 초기화되지 않은 변수들을 HTML 요소로 초기화
+    newTodoInput = document.getElementById('new-todo-title')
+    todoList = document.getElementById('todo-list')
+    todoElements = todoList.children
+    filters = document.querySelector('ul.filters')
+    filterAll = filters.querySelector('li a.all')
+    filterActive = filters.querySelector('li a.active')
+    filterCompleted = filters.querySelector('li a.completed')
+    // 해당 변수들에 대한 적절한 이벤트 처리기 등록
+    newTodoInput.addEventListener('keydown', addNewTodo)
+    filterAll.addEventListener('click', filterAllViewChange)
+    filterActive.addEventListener('click', filterActiveViewChange)
+    filterCompleted.addEventListener('click', filterCompletedViewChange)
+    filterContainer = [filterAll, filterActive, filterCompleted]
+    selectedFilter = filterAll
+
+    // 저장된 할 일 항목이 있는지 확인, 있다면 불러오고 없다면 초기화.
+    todos = ToDoElementStorage.getItem(KEYWORD)
+    if(todos == null){
+        todos = JSON.stringify([])
+        ToDoElementStorage.setItem(KEYWORD, todos)
+    }
+    todoElementsNameArray = JSON.parse(todos)
+    // 저장된 할 일 항목 각각에 대해 할일 추가 로직 수행.
+    todoElementsNameArray.forEach(elementName =>
+        drawNewTodo(elementName, localStorage.getItem(elementName)=='true')
+    )
+    
+}
+
+// 사용자 입력으로 새로운 할 일이 추가되는 함수
+function addNewTodo(event){
+    // 기본적인 예외 처리(공백 문자열, 중복 할 일 등)
+    if (event.keyCode != ENTER_KEYCODE) return
+    text = newTodoInput.value.trimStart().trimEnd()
+    if(text.length == 0){
+        newTodoInput.focus()
+        return;
+    }
```

Comment:
```suggestion
    if (event.keyCode === ENTER_KEYCODE && text.length) {
       text = newTodoInput.value.trimStart().trimEnd()
       newTodoInput.focus()
       return;
    }
```

이 부분은 조건문을 하나로 합쳐서 작성하는게 가독성이 더 좋을거 같습니다!

Reply:
불필요한 조건문 분리였던 것 같습니다. 하나로 병합해야겠네요.

review#2 커밋에서 반영되었습니다. 감사합니다.